app/src/main/java/com/example/fundatecheroes2/character/presentation/NewCharacterViewModel.kt

```diff
@@ -33,33 +38,47 @@ class NewCharacterViewModel : ViewModel() {
             state.value = ViewState.ShowErrorUrl
             return
         }
-        if(heroiVilao.equals("Herói ou Vilão?")){
+        if (heroiVilao.equals("Herói ou Vilão?")) {
             state.value = ViewState.ShowErrorHeroiVilao
             return
         }
         //digitar caractere / no android ainda caía aqui
         /*if (!aniversario.contains("/")) {
             state.value = ViewState.ShowErrorDate
-        }*/ else {
+        }*/
+        else {
             val character = Character(nome, url, descricao, heroiVilao, idade, aniversario)
-            val characterString = moshi.adapter(Character::class.java)
-                .toJson(character)
+            listCharacter = puxarLista(preferences)
+            listCharacter?.add(character)
+            val listString = moshi.adapter(MutableList::class.java)
+                .toJson(listCharacter)
 
-            salvar(characterString)
+            salvar(preferences, listString)
             state.value = ViewState.ShowSuccess
         }
     }
+    private fun puxarLista(preferences: SharedPreferences): MutableList<Character>? {
+            val characterListString = preferences.getString("characterList", "")
+            if(characterListString.isNullOrEmpty()){
+                return mutableListOf()
+            }
+        else {
+                val listType =
+                    Types.newParameterizedType(MutableList::class.java, Character::class.java)
+                val adapter: JsonAdapter<MutableList<Character>> = moshi.adapter(listType)
+                return adapter.fromJson(characterListString)!!
```

Comment:
O ideal seria não ter esse force no null no final, poderiamos retornar uma lista vazia
```suggestion
                return adapter.fromJson(characterListString).orEmpty()
```